Lesson2/Homework.py

```diff
@@ -0,0 +1,58 @@
+def parse(string_to_parse):
+    try:
+        string_without_domain = string_to_parse.split('?')[1]
+    except IndexError:
+        # URL не имеет параметров
+        return {}
+    # Отбросим якорь (#), если он есть
+    if string_without_domain.find("#") != -1:
+        string_without_domain = string_without_domain.split('#')[0]
+    step1 = string_without_domain.replace("=", " ")
+    step2 = step1.replace("&", " ")
+    step3 = step2.split()
+    try:
+        output = {step3[i]: step3[i + 1] for i in range(0, len(step3), 2)}
+    except IndexError:
+        # Некорректный URL!
+        print("Некорректный URL!")
+        return -1
+    return output
+
+def parse_cookie(string_to_parse):
+    output = {}
+    # Проверим достаточно ли симолов в строке и есть ли символ "="
+    while len(string_to_parse) > 2 and any(char == '=' for char in string_to_parse):
+            delimiter1 = string_to_parse.find("=")
+            param = string_to_parse[0:delimiter1]
+            delimiter2 = string_to_parse.find(";")
+            value = string_to_parse[delimiter1+1:delimiter2]
+            if (len(param) == 0 or len(value) == 0):
```

Comment:
`if (len(param) == 0 or len(value) == 0):` -> `if not len(param) or not len(value):`